tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -648,7 +647,14 @@ private void multiLevelNestedActivity(CompletionType how, int nestedCnt) throws
              * - C2 is completed and then compensated
              * - C3, ... are completed
              */
-            lraClient.cancelLRA(uris[1]); // compensate the first nested LRA
+
+            // compensate the first nested LRA in the enlisted resource
+            tckSuiteTarget
+                .path(LRA_RESOURCE_PATH).path(CANCEL_PATH)
+                .request()
+                .header(LRA_HTTP_CONTEXT_HEADER, uris[1])
```

Comment:
We call it without specifying it as a 'nested call' (not using LRA_HTTP_PARENT_CONTEXT_HEADER)

So the first invocation is as nested, the second is a top level. For me this means that this are 2 different situations and thus another participant is created.

Reply:
That shouldn't be the case, because uris[1] was created as a nested LRA so the implementation must know that it is nested. This is an end-user call so I would like to avoid adding a requirement for including parent context.

Reply:
You are correct @xstefank in that we added the parent context header so that the implementation can provide it during participant callbacks (Compensate, Complete, Status and Forget).

But @rdebusscher does make a good point since the [text we added to the LRA annotation](https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java#L117) to describe the header could be interpreted in the way Rudy suggests (even though we did not intend for it to be interpreted in that way). So can we update the LRA annotation text to be more specific. A small change to the first 3 lines of that javadoc would cover it, something like the following:
 
``` java
    /**
     * When an implementation of the LRA specification invokes any of the participant
     * callbacks (namely {@link Compensate}, {@link Complete}, {@link Status} and
     * {@link Forget}) in the context of a nested LRA it must ensure that
     * the parent LRA is made available via an HTTP header field with the
     * following name. The value contains the parent LRA id associated with
     * the HTTP request/response and is of type {@link java.net.URI}.
     */
```

Reply:
@mmusgrov @rdebusscher included this suggestion.

Reply:
Thanks. @rdebusscher does this update to the @LRA javadoc resolve your issue?

Reply:
Updated JavaDoc is about the Participants callback. But my comment was mainly around calling a Rest endpoint with just the LRAid (not the ParentLRAid although it is supposed to be nested)

And still, I have a bigger problem of calling a Nested Participant again which is already Closed.

But it seems I'm the only one.

Reply:
@rdebusscher I don't like calling the nested participant again either. I think the nested processing could be difficult to grasp from the user perspective. The responsibility of handling the details of the nested processing are more with user now (but he has enough information to do it safely).
The gitter discussion contains the multiple arguments from different angles https://gitter.im/eclipse/microprofile-lra?at=5e43e045f3718e705df97aff .

While the LRA model was designed that way the change would mean breaking it just before 1.0 release.

I would be happy if there is mood to open the discussion about nested processing again and find a way how to simplify it (here especially from the user perspective).
But I ended up in state of mind to open it after 1.0 (like a topic for the LRA 2.0)

Reply:
If a lot of people are not confident that we are doing nested LRAs correctly, have a good understanding of all the scenarios possible with all the variations and all the implications. 

We should not leave it as it is and revise it in 2.0, but just remove.